sstables/compaction_manager.cc

```diff
@@ -515,6 +515,8 @@ inline bool compaction_manager::maybe_stop_on_error(future<> f, stop_iteration w
 }
 
 void compaction_manager::submit(column_family* cf) {
+    if (cf->is_auto_compaction_disabled_by_user()) return;
```

Comment:
I think it's better to move this test to table::do_trigger_compaction(), so its impl becomes;

if (!_compaction_disabled && !_compaction_disabled_by_user) {
    _compaction_manager.submit(this);
}
I'd also rename _compaction_disabled_by_user to _minor/auto_compaction_disabled_by_user. As you know, there are many compaction types and not specifying the compaction type means that the flag should be enforced on all compactions types like major which is not true.

Reply:
but then anyone who calls the `compaction_manager::submit()` directly will be able to bypass this check. is it fine?

Reply:
There are no calls to other calls to submit aside from the one raphael pointed out.

Reply:
however @raphaelsc I agree with @sitano 's placement of the check.

trigger_compaction() will not send any more compactions, but if there are compactions queued but not started for this CF, we don't want to start them either.

Reply:
If you will look at the abstraction layers alone, the only valid place to place it is in CM. Even if there are no other callers currently. (it is simple to prove that from the opposite: if eg there were sites that were making submit() calls, eg the REST API can do calls directly to CM.Submit() and the tests sometimes call submit(), they would not respect this check, thus we either placed it wrong in the table or we intentionally wanted to limit its scope. but we didnt want either of those) (placing it in the table could be the right thing if each table had its own private instance of the CM, not the global one as we have it, so its exposed to the world, and the compaction control is available to everyone skipping the tables API whatsoever) 

Indeed, there are few usages of `submit()` from the CM itself, like: `compaction_submission_callback` and `postponed_compactions_reevaluation`. I am sure we don't want to start postponed compactions if user have disabled them.